src/controllers/api-key.js

```diff
@@ -10,82 +10,259 @@
+
+      const apiKeyEntity = await dataAccess.createNewApiKey({
+        id: crypto.randomUUID(),
+        name: data.name,
+        scopes,
+        createdAt: new Date().toISOString(),
+        expiresAt: expiresAt.toISOString(),
+        imsOrgId,
+        imsUserId,
+        hashedApiKey,
+      });
+
+      return createResponse(ApiKeyDto.toJSON(apiKeyEntity, apiKey), STATUS_CREATED);
+    } catch (error) {
+      log.error(`Failed to create a new api key: ${error.message}`);
       return createErrorResponse(error);
     }
-    return createResponse({}, 501);
   }
 
   /**
    * Delete an API key.
    * @param {Object} requestContext - Context of the request.
-   * @returns {Promise<Response>} 204 No Content.
+   * @returns {Promise<Response>} - 204 No Content.
    */
   async function deleteApiKey(requestContext) {
     const { pathInfo: { headers }, params: { id } } = requestContext;
-    const imsOrgId = headers['x-ims-gw-org-id'];
+    const imsOrgId = headers['x-gw-ims-org-id'];
+
+    try {
+      const imsUserToken = getImsUserToken(headers);
+      await validateImsOrgId(imsOrgId, imsUserToken);
+      const apiKeyEntity = await dataAccess.getApiKeyById(id);
+      const { authInfo: { profile } } = attributes;
 
-    validateImsOrgId(imsOrgId);
-    if (id) {
-      const error = new ErrorWithStatusCode('Delete API key not implemented', 501);
+      // Currently the email is assigned as the imsUserId
+      const imsUserId = profile.email;
+      if (!apiKeyEntity
+          || apiKeyEntity.getImsUserId() !== imsUserId || apiKeyEntity.getImsOrgId() !== imsOrgId) {
+        throw new ErrorWithStatusCode('Invalid request: API key not found', STATUS_NOT_FOUND);
+      }
+
+      apiKeyEntity.updateDeletedAt(new Date().toISOString());
+
+      await dataAccess.updateApiKey(apiKeyEntity);
+      return createResponse({}, STATUS_NO_CONTENT);
+    } catch (error) {
+      log.error(`Failed to delete the api key with id: ${id} - ${error.message}`);
       return createErrorResponse(error);
     }
-    return createResponse({}, 501);
   }
 
   /**
-   * Retrieve an API key.
+   * Retrieve the API keys relating to a specific imsUserId and imsOrgId combination.
    * @param {Object} context - Context of the request.
-   * @returns {Promise<Response>} 200 OK with the list of ApiKey metadata.
+   * @returns {Promise<ApiKey[]>} - 200 OK with the list of ApiKey metadata.
    */
   async function getApiKeys(requestContext) {
     const { pathInfo: { headers } } = requestContext;
-    const imsOrgId = headers['x-ims-gw-org-id'];
+    const imsOrgId = headers['x-gw-ims-org-id'];
+
+    try {
+      const imsUserToken = getImsUserToken(headers);
+      await validateImsOrgId(imsOrgId, imsUserToken);
+      const { authInfo: { profile } } = attributes;
 
-    validateImsOrgId(imsOrgId);
-    return createResponse({}, 501);
+      // Currently the email is assigned as the imsUserId
+      const imsUserId = profile.email;
```

Comment:
Helper method could be used from here as well `getImsUserIdFromProfile(profile)`